CMakeLists.txt

```diff
@@ -0,0 +1,353 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(libcoap VERSION 4.3.0 LANGUAGES CXX C)
```

Comment:
Does it make sense to have CMakeLists.txt as CMakeLists.txt.in , so that `VERSION 4.3.0` could be `VERSION @PACKAGE_VERSION`@ so that the VERSION gets automatically updated?

Reply:
CMakeLists.txt name should stay as this name is recognized by IDEs and package managers

I’ll think about importing this from other files instead of having this here

Reply:
same as for LIBCOAP_API_VERSION